tests/test_core.py

```diff
@@ -284,16 +285,19 @@ def test_region_processing_skip_aggregation(model_name, region_names):
     )
     add_meta(exp)
 
-    obs = process(
-        test_df,
-        dsd := DataStructureDefinition(
-            TEST_DATA_DIR / "region_processing/skip_aggregation/dsd"
-        ),
-        processor=RegionProcessor.from_directory(
-            TEST_DATA_DIR / "region_processing/skip_aggregation/mappings", dsd
-        ),
+    dsd = DataStructureDefinition(
+        TEST_DATA_DIR / "region_processing/skip_aggregation/dsd"
     )
-    assert_iamframe_equal(obs, exp)
+    processor = RegionProcessor.from_directory(
+        TEST_DATA_DIR / "region_processing/skip_aggregation/mappings", dsd
+    )
+    if model_name == "model_b":
```

Comment:
I think this makes the test rather hard to read having to follow those two different branches. I'd suggest, in analogy to many other tests in nomenclature, to split this into two tests. One that expects a certain data frame back and the other one that expects the error.